sklearn/datasets/_base.py

```diff
@@ -264,22 +269,47 @@ def load_data(module_path, data_file_name):
         A 1D array containing the names of the classifications. For example
         target_names[0] is the name of the target[0] class.
     """
-    with open(join(module_path, "data", data_file_name)) as csv_file:
-        data_file = csv.reader(csv_file)
-        temp = next(data_file)
-        n_samples = int(temp[0])
-        n_features = int(temp[1])
-        target_names = np.array(temp[2:])
-        data = np.empty((n_samples, n_features))
-        target = np.empty((n_samples,), dtype=int)
-
-        for i, ir in enumerate(data_file):
-            data[i] = np.asarray(ir[:-1], dtype=np.float64)
-            target[i] = np.asarray(ir[-1], dtype=int)
+    with importlib_resources.path(data_module, data_file_name) as data_path:
+        with open(data_path) as csv_file:
```

Comment:
we should not `open(data_path)` assuming `data_path` is a valid filesystem path. Instead we should use [importlib.resource.open_binary](https://docs.python.org/3/library/importlib.html#importlib.resources.open_binary) or [importlib.resource.open_text](https://docs.python.org/3/library/importlib.html#importlib.resources.open_text) and pass the result to the `csv_reader`. I think for the `csv_reader` we need to open the resource in text mode, and probably hard-code `encoding="utf-8"` to avoid platform-dependent behaviors.